### DISCUSSION TITLE
Book nr. 1 : Focal Length

### DISCUSSION THREAD

**Discussion by lucabicego** [RayTracing/raytracing.github.io#1412] [Q&A]
Hi,
i did some test on the code that you find on chapter nr. 5.2 of the book 'Ray Tracing in One Weekend'. 
I used some values of the focal length to better understand what happen to the final picture. I used the values 0.5, 1 and 2. What i noticed is that with a focal length of 0.5 the picture of the sphere is little (see the first sphere on the left of the following picture) and if i use a focal length of 2 the picture of the sphere is bigger (see the sphere on the right). I tought to obtain the oposite ie with a little value of the focal length to obtai a bigger picture of the sphere. I don't understand why i get such a behaviour.
Could someone try to explain me it ?
Thanks.
Luca Bicego

<img width="610" alt="2024-02-27_14-15-09" src="https://github.com/RayTracing/raytracing.github.io/assets/30547111/643366fd-0fde-4179-83af-b171db5bf316">

**Comment by rupsis** [comment]
Hi @lucabicego, 

First thing to note, this camera created here isn't physically accurate.
The Sphere's center sits at (0,0,-1), so it's -1 in the z direction (which is going into the viewport in this instance). 

```
 auto viewport_upper_left = camera_center - vec3(0, 0, focal_length) - viewport_u/2 - viewport_v/2;
 ``` 
This equation right here leaves us with negative focal_length, since all of the other vec3's don't have a Z component. Thus the focal length is -1, and the spheres center is -1. This frames the sphere nicely. 

If we "decrease" the focal length to 0.5, the calculated focal length is -0.5, meaning it's half the distance to the sphere, and thus the sphere appears further aways. The opposite is true when you "increase" the focal point to 2.0. The calculated value is -2, which is behind the sphere, and thus the sphere appears larger. 

Again, this modeled to be a real world camera, but it does follow the same patterns as a camera. I.e, a lower focal length the subject is further away, larger focal length the subject is closer:

![image](https://s.studiobinder.com/wp-content/uploads/2019/02/understanding-focal-length-Same-Distance.jpg)
(Image from [studiobinder.](https://www.studiobinder.com/blog/focal-length-camera-lenses-explained/))

**Reply by lucabicego** [reply]
Hi Ruspis, thank you for your answer.
Now it's clear.
Luca

**Comment by hollasch** [comment]
> This equation right here leaves us with negative focal_length

Focal length is an actual length/distance, and so is always positive. Direction is irrelevant. Put another way, the projection geometry is unchanged by moving or rotating the camera.

**Comment by hollasch** [comment]
@lucabicego — in its simplest terms, the camera is just a rectangular "viewport" and its corresponding projection point. Scene rays originate from the projection point through some arbitrary point inside the viewport. If the projection point is infinitely far away, then you can imagine all of the rays going through the viewport parallel to each other — the "beam" of all rays is as narrow as it can be, and only objects inside that beam will be visible. As the projection point moves closer to the viewport, the "beam" widens, capturing more and more of the scene on the other side of the viewport.

The _viewing angle_ is the resulting angle of this beam, so that a larger angle (wide angle) makes things rapidly appear smaller the further they are from the viewport. This corresponds to a short focal length. A long focal length correspondingly yields a narrow (zoom) viewing angle, so there's less of a size difference between near and distant objects.
